Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/StatusBean.java

```diff
@@ -168,16 +166,16 @@ public PieChartModel getEventPieChart() {
 
     private void refreshPieChart() {
         if (snapshot.getEvents().isEmpty()) {
-            List<String> serviceNames = NhincConstants.EVENT_LOGGING_SERVICE_NAME.getEventLoggingServiceForDisplay();
-            for (String serviceName : serviceNames) {
-                eventPieChart.getData().put(serviceName, 0);
-            }
             eventPieChart.setTitle(NO_DATA_FOUND);
-        } else {
-            for (EventCount event : snapshot.getEvents().values()) {
-                eventPieChart.getData().put(event.getEvent(), event.getTotal());
-            }
         }
+        List<String> serviceNames = NhincConstants.EVENT_LOGGING_SERVICE_NAME.getEventLoggingServiceForDisplay();
+        int count;
+        for (String serviceName : serviceNames) {
+            count = (snapshot.getEvents().get(serviceName) == null || snapshot.getEvents().get(serviceName).getTotal()
```

Comment:
Optional: Define variable for snapshot.getEvents().get(serviceName) so that it doesn't retrieve the list each time we compare.